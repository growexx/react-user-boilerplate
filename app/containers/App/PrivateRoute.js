/* eslint-disable space-before-function-paren */
import React from 'react';
import { Route, Redirect } from 'react-router';
import PropTypes from 'prop-types';
import { userExists } from '../../utils/userExists';

const PrivateRoute = mainProps => {
  const { component: Component, ...rest } = mainProps;

  return (
    <Route
      {...rest}
      render={props =>
        userExists() ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: '/login',
            }}
          />
        )
      }
    />
  );
};

PrivateRoute.propTypes = {
  component: PropTypes.func,
};

export default PrivateRoute;
