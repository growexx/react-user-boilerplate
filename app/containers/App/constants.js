/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

import {
  FAV_ICON_152_152,
  FAV_ICON_16_16,
  FAV_ICON_32_32,
  APPLE_TOUCH_ICON,
  FAV_ICON_192_192,
} from '../../images/favicons';
export const LOAD_REPOS = 'boilerplate/App/LOAD_REPOS';
export const LOAD_REPOS_SUCCESS = 'boilerplate/App/LOAD_REPOS_SUCCESS';
export const LOAD_REPOS_ERROR = 'boilerplate/App/LOAD_REPOS_ERROR';
export const LOAD_APP = 'boilerplate/App/LOAD_APP';
export const FAV_ICONS = [
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '512x512',
    href: FAV_ICON_152_152,
  },

  {
    rel: 'icon',
    type: 'image/png',
    sizes: '192x192',
    href: FAV_ICON_192_192,
  },
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '32x32',
    href: FAV_ICON_32_32,
  },
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '16x16',
    href: FAV_ICON_16_16,
  },
  {
    rel: 'apple-touch-icon',
    href: APPLE_TOUCH_ICON,
  },
];
