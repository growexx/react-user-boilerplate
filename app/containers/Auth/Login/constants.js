export const CHANGE_EMAIL = 'boilerplate/Login/CHANGE_EMAIL';
export const CHANGE_PASSWORD = 'boilerplate/Login/CHANGE_PASSWORD';
export const ERROR = 'boilerplate/Login/ERROR';
export const LOADING = 'boilerplate/Login/LOADING';
export const SUCCESS = 'boilerplate/Login/SUCCESS';
export const LOGIN = 'boilerplate/Login/LOGIN';
export const RESET = 'boilerplate/Login/RESET';
export const GOOGLE_LOGIN = 'boilerplate/Login/GOOGLELOGIN';
export const FACEBOOK_LOGIN = 'boilerplate/Login/FACEBOOKLOGIN';
