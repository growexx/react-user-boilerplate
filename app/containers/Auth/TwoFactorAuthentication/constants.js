/*
 *
 * TwoFactorAuthentication constants
 *
 */

export const CHANGE_VALUE = 'app/TwoFactorAuthentication/CHANGE_VALUE';
export const SUBMIT = 'app/TwoFactorAuthentication/SUBMIT';
export const FORM_KEY = 'twoFactorAuthentication';
export const TEST_OTP_VALUE = 123456;
export const TEST_OTP_VALUE_WITHOUT_LENGTH = 12345;
export const OTP_LENGTH = 6;
