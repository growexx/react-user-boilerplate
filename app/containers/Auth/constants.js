export const AUTH_TYPE = ['LOGIN', 'REGISTER'];

// API-ROUTES
export const API_URL = 'https://b0257f3aa104.ngrok.io';
export const AUTH = 'auth';
export const API_ENDPOINTS = {
  LOGIN: `${API_URL}/${AUTH}/signin`,
  LIST: 'https://randomuser.me/api/?results=5&inc=name,gender,email,nat&noinfo',
  LIST_AVATAR:
    'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
  EXPORT_CSV: '',
};
