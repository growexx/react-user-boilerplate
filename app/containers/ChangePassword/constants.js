/*
 *
 * ChangePassword constants
 *
 */
const PASSWORD_MUST_MATCH = 'Passwords do not match';
export const passwordsMustMatch = (value, allValues) =>
  value !== allValues.newPassword ? PASSWORD_MUST_MATCH : undefined;

export const CHANGE_CURRENT_PASSWORD =
  'boilerplate/ChangePassword/CHANGE_CURRENT_PASSWORD';
export const CHANGE_NEW_PASSWORD =
  'boilerplate/ChangePassword/CHANGE_NEW_PASSWORD';
export const CHANGE_CONFIRM_NEW_PASSWORD =
  'boilerplate/ChangePassword/CHANGE_CONFIRM_NEW_PASSWORD';
export const LOADING = 'boilerplate/ChangePassword/LOADING';
export const SUBMIT_DATA = 'boilerplate/ChangePassword/SUBMIT_DATA';
export const FORM_KEY = 'ChangePassword';
