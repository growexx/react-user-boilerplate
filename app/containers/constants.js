// Control Bit
export const IS_DEMO_CODE = true;

// App-Routes
export const ROUTES = {
  HOME: '/',
  GITHUB_SEARCH: '/github-search',
  FONT_AWESOME: '/font-awesome',
  LOGIN: '/login',
  REGISTER: '/register',
  PROFILE: '/profile',
  LOGOUT: '/logout',
  LOADER: '/loader',
  EXPORT_DATA: '/export-data',
  USERS: '/users',
  CHARTS: '/charts',
  TEST_ADMIN_PAGE: '/admin',
  UNAUTHORIZED: '/403',
  NUMERAL_CONVERTER: '/number-converter-demo',
  SAMPLE_FORM: '/sample-form',
  CHANGE_PASSWORD: '/change-password',
  TWO_FACTOR_AUTHENTICATION: '/two-factor-authentication',
  FORGOT_PASSWORD: '/forgot-password',
  MULTI_TAB_SUPPORT: '/multi-tab-support',
  PRODUCTS: '/products',
};

// API-ROUTES
export const { API_URL } = process.env;
export const AUTH = 'auth';
export const API_ENDPOINTS = {
  LOGIN: `${API_URL}/${AUTH}/signin`,
  LIST: 'https://randomuser.me/api/?results=5&inc=name,gender,email,nat&noinfo',
  LIST_AVATAR:
    'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
  EXPORT_CSV: '',
  IMAGE_UPLOAD: 'https://www.mocky.io/v2/5cc8019d300000980a055e76',
  FORGOT_PASSWORD: `${API_URL}/${AUTH}/forgetpassword`,
  USERS: `${API_URL}/users`,
};

// Table Pagination default
export const DEFAULT_PAGE = 1;
export const DEFAULT_LIMIT = 10;

/**
 * Roles for system
 * ACL from frontend
 * This configuration is being used in RoleMiddleware
 */
// Note: Define role in this constant
export const ROLES = {
  USER: 1,
  ADMIN: 10,
};

// Note: Add Routes which needs to check role before it's access
export const RESTRICTED_ROUTES = [ROUTES.PROFILE, ROUTES.TEST_ADMIN_PAGE];

// Note: Define role against all accessible routes
export const ROLE_BASED_ROUTE_ACCESS = {
  [ROLES.USER]: [ROUTES.PROFILE],
  [ROLES.ADMIN]: [ROUTES.TEST_ADMIN_PAGE],
};

// Note: Role based default routes for redirection
export const ROLE_BASED_DEFAULT_ROUTE = {
  [ROLES.USER]: ROUTES.HOME,
};

/**
 * Controls which sidebar entries shown to this user
 */
// Sidebar role based update
export const ROLE_BASED_SIDEBAR_MENU = {
  [ROLES.USER]: [],
  [ROLES.ADMIN]: [ROUTES.TEST_ADMIN_PAGE],
};

export const DEFAULT_PAGE_NO = 1;
export const DEFAULT_SIZE = 10;
export const SORTING = {
  ASC: 1,
  DESC: -1,
  ascend: 1,
  descend: -1,
};
/**
 *
 * @param {string} order  ['ascend','descend']
 * @returns
 */
export const GET_SORT_ORDER = order => {
  switch (order) {
    case 'ascend':
      return SORTING.ASC;
    case 'descend':
      return SORTING.DESC;
    default:
      return SORTING.DESC;
  }
};

export const GET_DEFAULT_PAGINATION = () => ({
  pageSize: DEFAULT_SIZE,
  current: DEFAULT_PAGE_NO,
  total: 0,
});

export const GENERIC_MOMENT_DATE_FORMAT = 'YYYY-MM-DD';
export const FULL_GENERIC_MOMENT_DATE_FORMAT = 'YYYY-MM-DD hh:mm:ss a';
export const CHANNEL_NAME = 'DATA_CHANNEL_NAME';
