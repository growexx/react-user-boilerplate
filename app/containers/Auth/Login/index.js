/**
 *
 * Login
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import { Form, Input, Button, notification } from 'antd';
import {
  FacebookFilled,
  GoogleOutlined,
  WindowsFilled,
} from '@ant-design/icons';
import { connect } from 'react-redux';
import { compose } from 'redux';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { useInjectReducer } from 'utils/injectReducer';
import { useInjectSaga } from 'utils/injectSaga';
import {
  makeSelectEmail,
  makeSelectLoading,
  makeSelectError,
  makeSelectPassword,
} from 'containers/Auth/Login/selectors';
import messages from './messages';
import { StyledLogin } from './StyledLogin';
import { makeSelectSuccess } from './selectors';
import { changeEmail, changePassword, fireLogin } from './actions';
import reducer from './reducer';
import saga from './saga';
import { StyledAuthContainer } from '../StyledAuthContainer';
import AuthSideContainer from '../index';
import { AUTH_TYPE } from '../constants';
import { ROUTES } from '../../constants';

const key = 'login';

const showNotification = () => {
  notification.open({
    message: <FormattedMessage {...messages.notificationToast} />,
  });
};
export function Login({
  loading,
  onSignIn,
  email,
  password,
  error,
  onChangeEmail,
  onChangePassword,
}) {
  useInjectReducer({ key, reducer });
  useInjectSaga({ key, saga });
  return (
    <Form onFinish={onSignIn}>
      <StyledAuthContainer>
        <Helmet>
          <title>Login</title>
          <meta name="description" content="Description of Login" />
        </Helmet>
        <AuthSideContainer authType={AUTH_TYPE[0]} />
        <StyledLogin>
          <p className="createAccount">
            <FormattedMessage {...messages.accountDetails} />
          </p>
          <div className="LoginSubContainer">
            <div className="socialIcons">
              <FacebookFilled />
              <GoogleOutlined />
              <WindowsFilled />
            </div>
            <p className="emailLogin">
              <FormattedMessage {...messages.emailLogin} />
            </p>
            <div className="accountData">
              <Form.Item
                name="email"
                rules={[
                  {
                    type: 'email',
                    message: <FormattedMessage {...messages.validEmail} />,
                  },
                  {
                    required: true,
                    message: (
                      <FormattedMessage {...messages.emailRequiredMessage} />
                    ),
                  },
                ]}
              >
                <Input
                  placeholder="Email"
                  onChange={onChangeEmail}
                  type="email"
                  value={email}
                />
              </Form.Item>
              <Form.Item
                name="password"
                rules={[
                  {
                    required: true,
                    message: (
                      <FormattedMessage {...messages.passwordRequiredMessage} />
                    ),
                  },
                ]}
              >
                <Input.Password
                  value={password}
                  placeholder="Password"
                  onChange={onChangePassword}
                  type="password"
                />
              </Form.Item>
            </div>
            <Form.Item>
              <Button loading={loading} htmlType="submit">
                <FormattedMessage {...messages.signIn} />
              </Button>
            </Form.Item>
          </div>
          <Link to={ROUTES.FORGOT_PASSWORD}>
            <FormattedMessage {...messages.forgotPassword} />
          </Link>
        </StyledLogin>
        {error === true && showNotification()}
      </StyledAuthContainer>
    </Form>
  );
}

Login.propTypes = {
  history: PropTypes.object,
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  success: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  onSignIn: PropTypes.func,
  email: PropTypes.string,
  password: PropTypes.string,
  onChangePassword: PropTypes.func,
  onChangeEmail: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  email: makeSelectEmail(),
  password: makeSelectPassword(),
  loading: makeSelectLoading(),
  error: makeSelectError(),
  success: makeSelectSuccess(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onChangeEmail: evt => dispatch(changeEmail(evt.target.value)),
    onChangePassword: evt => dispatch(changePassword(evt.target.value)),
    onSignIn: evt => {
      if (evt !== undefined && evt.preventDefault) evt.preventDefault();
      dispatch(fireLogin());
    },
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(Login);
