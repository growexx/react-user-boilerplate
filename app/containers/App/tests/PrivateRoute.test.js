import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import { TOKEN_KEY } from 'utils/constants';
import StorageService from 'utils/StorageService';
import Login from 'containers/Auth/Login/Loadable';
import configureStore from '../../../configureStore';
import PrivateRoute from '../PrivateRoute';

jest.mock('utils/Helper');

let store;
const tokenValue = 'test token';
const props = {
  component: Login,
  path: '/admin',
  showError: true,
};
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <PrivateRoute {...props} />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

const login = () => StorageService.set(TOKEN_KEY, tokenValue);

describe('<PrivateRoute />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
    login();
  });
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('show Error false', () => {
    props.showError = false;
    const { container } = componentWrapper();
    expect(container).toBeTruthy();
  });
});
