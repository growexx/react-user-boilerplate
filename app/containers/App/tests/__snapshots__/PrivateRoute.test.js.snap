// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> should render and match the snapshot 1`] = `null`;
